src/api_proxy/service_control/request_builder.cc

```diff
@@ -1371,62 +1371,82 @@ Status RequestBuilder::ConvertCheckResponse(
   const CheckError& error = check_response.check_errors(0);
   switch (error.code()) {
     case CheckError::NOT_FOUND:
+      check_response_info->error_type = CheckResponseErrorType::CONSUMER_ERROR;
       return Status(Code::INVALID_ARGUMENT,
                     "Client project not found. Please pass a valid project.");
     case CheckError::RESOURCE_EXHAUSTED:
```

Comment:
Are you going to add counter denied_consumer_quota  here

Here from the feature request
service_conrol.denied_consumer_quota
       + Check response with RESOURCE_EXHAUSTED error code

Reply:
Yes, I will do that in another PR, the feature request is not done yet. 

I have some questions I need to clarify first. Specifically: 
- Shouldn't this counter be incremented in `AllocateQuota` calls, not `Check` calls?
- What is the difference between `CheckResponse::RESOURCE_EXHUASTED` and  `AllocateQuotaResponse::RESOURCE_EXHUASTED`?

Reply:
Good question.   In ESF used CheckRequest,  it has quota request.  so CheckResponse::RESOURCE_EXHUASTED is for ESF.    For ESP,  we don't use it,  we call separate API AllocateQuota,   so we should only get AllocateQuotaResponse::RESOURCE_EXHUASTED,  not CheckResponse::RESOURCE_EXHUASTED

Reply:
Makes sense, thanks